docs(vault): document local field for OAuth Resource Server profiles (#3150)
hashicorp/web-unified-docs: content/vault/v2.x/content/docs/ai/oauth-server/index.mdx

--------------------------------- | -----------
`name`                            | Unique name for the profile within the namespace. Used as the path identifier in API operations.
`config_id`                       | Stable unique identifier generated by Vault when the profile is created.
`issuer_id`                       | The expected `iss` claim value. Must be unique within its locality (local or global) in the namespace. Required on create.

> `local`                           | If `true`, the profile is stored in cluster-local storage and is never replicated to other clusters. Defaults to `false`. Immutable after creation.

Reviewer: ```suggestion
`local`                           | If `true`, Vault stores the profile in cluster-local storage and never replicates it to other clusters. Defaults to `false`. Immutable after creation.
```

HMAC algorithms, including HS256, HS384, and HS512, opaque tokens, and JWE,
or JSON Web Encryption, are not supported.

## Local profiles

By default, profiles are **global**: Vault stores them in replicated storage and

> propagates them to all performance secondary clusters automatically.

Reviewer: ```suggestion
By default, profiles are **global**. Vault stores profiles in replicated storage and
propagates them to all performance secondary clusters automatically.
```
Style correction: use complete prose sentences in page text, avoid cross-sentence pronouns

Set `local=true` when creating a profile to make it **local** to the cluster

> where it is created. Local profiles:

Reviewer: ```suggestion
Set `local=true` when you want to create a profile and make it **local** to the cluster
where you create it. Local profiles:
```
Style correction: write in active voice

- Are stored in cluster-local storage and never enter the replication WAL.
- Exist only on the cluster where they were created — they are not visible on

>   other clusters.

Reviewer: ```suggestion
- Exist only on the cluster where they were created. Vault does not make local
  profiles visible on other clusters.
```
Style correction: avoid em-dashes

- Can be created, updated, and deleted directly on a performance secondary
  without write-forwarding to the primary.

The `local` field is immutable. To change a profile's locality, delete it and

> recreate it with the desired `local` value.

Reviewer: ```suggestion
The `local` field is immutable. To change the locality of a profile, delete it and
recreate it with the desired `local` value.
```
Style correction: avoid possessives

### When to use local profiles

Use local profiles when you need to configure trust for an issuer that is
specific to one cluster and should not propagate elsewhere, for example, a

> private authorization server that operates only in a single region.

Reviewer: ```suggestion
specific to one cluster and should not propagate elsewhere. For example, when you
have a private authorization server that operates only in a single region.
```
Style correction: use simple, complete sentences

### Local profile precedence during JWT validation

When Vault receives a JWT, it resolves the profile by the token's `iss` claim.
If both a local and a global profile share the same `issuer_id`, Vault uses the
**local profile** to validate the token. This lets you override a globally

> replicated profile on a per-cluster basis.

Reviewer: ```suggestion
When Vault receives a JWT, it resolves the profile using the `iss` claim in the
issuer token. If both a local and a global profile share the same `issuer_id`,
Vault uses the **local profile** to validate the token. Using the local profile
lets you override a globally replicated profile on a per-cluster basis.
```
Style correction: avoid possessives, avoid "this" as a pronoun

If validation unexpectedly fails for tokens from an issuer that has a global
profile, check whether the cluster also has a local profile with the same
`issuer_id`. A local profile silently takes precedence over the global one, so
misconfigured key material or audiences in the local profile will cause
rejections even though the global profile appears correct. Use
`vault read sys/config/oauth-resource-server/<name>` and confirm the `local`
field on each profile to identify which profile Vault is actually using on that

> cluster.

Reviewer: ````suggestion
If validation unexpectedly fails for tokens from an issuer that has a global
profile, Vault checks whether the cluster has a local profile with the same
`issuer_id`. Local profiles silently take precedence over global profiles, so
misconfigured key material or audiences in the local profile will cause
rejections even when the global profile appears correct.

If you see unexpected rejections, use the `sys/config/oauth-resource-server/<name>`
endpoint to confirm the `local` field for each profile and confirm which profile
Vault is actually using on that cluster. For example:

```shell-session
$ vault read sys/config/oauth-resource-server/my-profile
```

````

## Issuer uniqueness

Each `issuer_id` must be unique **per locality**:

- Two local profiles in the same namespace cannot share an `issuer_id`.
- Two global profiles in the same namespace cannot share an `issuer_id`.
- A local profile and a global profile *may* share an `issuer_id` — this is
  the intended configuration when a local profile is used to override a global

>   one on a specific cluster.

Reviewer: ```suggestion
- A local profile and a global profile *may* share an `issuer_id`. Vault expects
  shared issuer IDs when you use a local profile to override a global profile on
  a specific cluster.
```
Style correction: write in active voice, avoid "this" as a pronoun
